### ISSUE TITLE
can we ignore a metric which does not belong to the resource while calling listWithResponse

### ISSUE THREAD

**Issue by yogeshprasad** [Azure/azure-sdk-for-java#33240]
**Query/Question**
Hi Team,
I am trying to write a client application which gets the metrics for the resources and to save some API calls I am caching the metric definition by a resource type assuming all the resources under that resource type will have the same metrics.  Later I found that's not the case some resources are having a different set of metrics even though they are under the same resource type. 
**Can someone confirm my understanding is correct?** 

and when I am calling the method **listWithResponse(String var1, String var2, Duration var3, String var4, String var5, Integer var6, String var7, String var8, ResultType var9, String var10, Context var11)** to get the metrics with the additional metric which does not belongs to the resource it is throwing below error:

```
com.azure.core.management.exception.ManagementException: Status code 400, "{"code":"BadRequest","message":"Failed to find metric configuration for provider: Microsoft.Network, resource Type: publicIPAddresses, metric: VipAvailability, Valid metrics: PacketsInDDoS,PacketsDroppedDDoS,PacketsForwardedDDoS,TCPPacketsInDDoS,TCPPacketsDroppedDDoS,TCPPacketsForwardedDDoS,UDPPacketsInDDoS,UDPPacketsDroppedDDoS,UDPPacketsForwardedDDoS,BytesInDDoS,BytesDroppedDDoS,BytesForwardedDDoS,TCPBytesInDDoS,TCPBytesDroppedDDoS,TCPBytesForwardedDDoS,UDPBytesInDDoS,UDPBytesDroppedDDoS,UDPBytesForwardedDDoS,IfUnderDDoSAttack,DDoSTriggerTCPPackets,DDoSTriggerUDPPackets,DDoSTriggerSYNPackets"}"
```
**Why can't we ignore a metric which does not belong to the resource?**

Basically, I don't want to make the API calls per resource to know the metric definition as this will cost when we will have a good amount of resources.

***Why is this not a Bug or a feature Request?***
I am new to this and there might be a way to handle this. will create an issue if I will get confirmation on this.

**Setup (please complete the following information if applicable):**
 This can be reproduced easily for the resource type: **Microsoft.Network/publicIPAddresses**
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Query Added
- [X] Setup information Added

**Comment by joshfree** [comment]
@weidongxu-microsoft could you please follow up?

**Comment by weidongxu-microsoft** [comment]
@yogeshprasad 

Do you mean the request of `VipAvailability` on some "publicIPAddresses" would be a success, but on some be a 400?

> Later I found that's not the case some resources are having a different set of metrics even though they are under the same resource type.
Can someone confirm my understanding is correct?

**Comment by weidongxu-microsoft** [comment]
At present I don't have a very good solution, as 400 is by backend REST API.

What I can suggest is cross check with current doc https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-supported
And try-catch the 400 and ignore them (code=BadRequest makes it hard to filter for this particular case)

**Comment by weidongxu-microsoft** [comment]
@XiaofeiCao 
Do you know whether we've got other APIs to query metrics of a resource? E.g. some data-plane monitor lib, some GraphQL, Azure Grafana?

**Comment by yogeshprasad** [comment]
@weidongxu-microsoft 
Yes that is true for some instances it is passing and for some, it is not when I ask for the same metrics.
> Do you mean the request of VipAvailability on some instances of "publicIPAddresses" would be a success, but on some be a 400? (what's the difference of the 2 kind of resources with same type?)

and this is not just with one service I have seen this with other services also. here at least I am looking to not reject the entire API call as we are sending a batch of metrics to get the data points. probably we can get some message in the error field which can tell that some metric does not belong to this resource.

**Comment by weidongxu-microsoft** [comment]
There is a data-plane lib "azure-monitor-query". But it seems uses the same endpoint and same api-verion. I guess it will have same issue...
https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/monitor/azure-monitor-query#metrics-query

**Comment by yogeshprasad** [comment]
any leads on the right place where this can be reported ?

**Comment by weidongxu-microsoft** [comment]
I am adding backend to the issue.

**Comment by KevinBlasko** [comment]
This behavior is expected.  Some metrics are only available for certain SKUs of a particular resource type.  For publicIPAddresses in particular, different metrics will be available based on the sku.name property on a particular resource's metadata.  VipAvailability, ByteCount, PacketCount, and SynCount are only available on Standard skus.

If you don't want to interrogate metricdefinitions for each publicIPAddress, you should consider making an [Azure Resource Graph](https://learn.microsoft.com/en-us/azure/governance/resource-graph/overview) query to make a list of all Standard sku resources and query them separately from non Standard sku resources.  Here's a quick sample query you can use as a starting point.  Just flip the sku.name == "Standard" to sku.name != "Standard" to get the other list.

`Resources
| where type =~ 'microsoft.network/publicIPAddresses'
| where sku.name == "Standard"
| limit 100`

**Comment by yogeshprasad** [comment]
Thanks, @KevinBlasko, yes this seems to be working, but there are some resources without any SKU, can that also be considered as non-Standard?

**Comment by yogeshprasad** [comment]
Closing this as I am able to fix the issue by using the SKU name with the resource type to cache the metric definition.
Thanks, Azure Team
